File: src/Z.Core/System.String/System.Web.HttpUtility/String.UrlPathEncode.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;
using System.Web;

public static partial class Extensions
{
    /// <summary>
    ///     Encodes the path portion of a URL string for reliable HTTP transmission from the Web server to a client.
    /// </summary>
    /// <param name="str">The text to encode.</param>
    /// <returns>The encoded text.</returns>
    public static String UrlPathEncode(this String str)
    {
        return HttpUtility.UrlPathEncode(str);
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/System.Web.HttpUtility/String.UrlPathEncode.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Web

Public Module Extensions_580

	''' <summary>
	'''     Encodes the path portion of a URL string for reliable HTTP transmission from the Web server to a client.
	''' </summary>
	''' <param name="str">The text to encode.</param>
	''' <returns>The encoded text.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function UrlPathEncode(str As [String]) As [String]
		Return HttpUtility.UrlPathEncode(str)
	End Function
End Module


